reader.go

```diff
@@ -128,21 +148,29 @@ func (rd *Reader) skipUnread() error {
 	return err
 }
 
-func (rd *Reader) readHeader() (*Header, error) {
+// Next skips to the next file in the archive file.
+// Returns a Header which contains the metadata about the
+// file in the archive. io.EOF is returned at the end of the input.
+func (rd *Reader) Next() (*Header, error) {
+	err := rd.skipUnread()
+	if err != nil {
+		return nil, err
+	}
+
 	headerBuf := make([]byte, HEADER_BYTE_SIZE)
 	if _, err := io.ReadFull(rd.r, headerBuf); err != nil {
 		return nil, err
 	}
 
-	header := new(Header)
 	s := slicer(headerBuf)
-
-	header.Name = rd.string(s.next(16))
-	header.ModTime = time.Unix(rd.numeric(s.next(12)), 0)
-	header.Uid = int(rd.numeric(s.next(6)))
-	header.Gid = int(rd.numeric(s.next(6)))
-	header.Mode = rd.octal(s.next(8))
-	header.Size = rd.numeric(s.next(10))
+	header := &Header{
+		Name:    rd.string(s.next(16)),
+		ModTime: time.Unix(rd.numeric(s.next(12)), 0),
+		Uid:     int(rd.numeric(s.next(6))),
+		Gid:     int(rd.numeric(s.next(6))),
+		Mode:    rd.octal(s.next(8)),
+		Size:    rd.numeric(s.next(10)),
+	}
```

Comment:
Just to check: Does Go execute this line by line, or in field definition order? Is it worth a comment saying that changing the order will make it explode?